lab-allie/controller/album-controller.js

```diff
@@ -0,0 +1,47 @@
+'use strict';
+
+const createError = require('http-errors');
+const Promise = require('bluebird');
+const Album = require('../model/album.js');
+
+module.exports = exports = {};
+
+exports.createAlbum = function(req, res, album) {
+  if(!album) return Promise.reject(createError(400, 'Album required'));
+  
+  new Album(req.body).save()
+  .then(album => res.json(album))
+  .catch(err => res.status(400).send(err.message));
+};
+
+exports.fetchAlbum = function(id, res) {
+  if(!id) return Promise.reject(createError(400, 'ID required'));
+  
+  return Album.findById(id)
+  .then(album => {
+    res.json(album);
+  })
+  .catch(err => res.status(400).send(err.message));
+};
+
+exports.fetchAllAlbums = function(res) {
+  return Album.find()
+  .then(album => res.json(album))
+  .catch(err => res.status(400).send(err.message));
+};
+
+exports.updateAlbum = function(req, res, id) {
+  if(!id) return Promise.reject(createError(400, 'ID required'));
+  
+  Album.findOneAndUpdate(id, req.body, {new: true})
```

Comment:
yes! :)  mongoose can be weird to filter through and find the proper arguments to pass in for updating. well done.